spec/operators/switchFirst-spec.js

```diff
@@ -0,0 +1,159 @@
+/* globals describe, it, expect, hot, cold, expectObservable, expectSubscriptions */
+var Rx = require('../../dist/cjs/Rx');
+var Promise = require('promise');
+
+var Observable = Rx.Observable;
+var immediateScheduler = Rx.Scheduler.immediate;
+
+describe('Observable.prototype.switchFirst()', function () {
+  it('should switch to first immediately-scheduled inner Observable', function () {
+    var e1 = cold( '(ab|)');
+    var e2 = cold( '(cd|)');
+    var expected = '(ab|)';
+    expectObservable(Observable.of(e1, e2).switchFirst()).toBe(expected);
+  });
```

Comment:
Can you add subscription tests also for these basic cases? Sometimes we discover bugs related to synchronous subscribe and unsubscribe `expected: (^!)` and sometimes unsubscription never happens (and should).